### DISCUSSION TITLE
Does bcc support calling kfuncs?

### DISCUSSION THREAD

**Discussion by thejh** [iovisor/bcc#5130] [General]
I'm trying to write BPF code that calls a kfunc (by kfunc I mean "kernel function callable from BPF", not "kernel function hooked by BPF"); but I can't figure out how to make that work. When I declare the kfunc as an extern function with `__attribute__((section(".kyms")))` and call it, I get a compiler error saying "cannot call non-static helper function". Is it just not possible to call kfuncs from code built with bcc, or am I doing it wrong?

**Comment by yonghong-song** [comment]
Sorry, bcc does not support kfunc. It need some effort to implement it. libbpf-tools supports kfunc.

**Comment by thejh** [comment]
Thanks for the information!
